File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        ans = [0,0];
        nummap = {}
        for i in range(0,len(nums)):
            if target - nums[i] in nummap:
                ans[0] = i
                ans[1] = nummap[target - nums[i]];
            else:
                nummap[nums[i]] = i
        return ans        

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] ans = new int[2];
        //Brute Force is 2 loops
        //Optimized is HashMap
        HashMap<Integer,Integer> map = new HashMap<>();
        for(int i = 0; i<nums.length; i++){
            if(map.containsKey(target - nums[i])){
                ans[0] = i;
                ans[1] = map.get(target - nums[i]);
            }
            else map.put(nums[i] , i);
        }
        return ans;
    }
}